src/sg_client.rs

```diff
@@ -105,6 +130,57 @@ impl SGClient {
             Ok(res.text()?)
         }
     }
+
+    /// Sends a messages through the SendGrid API. It takes a Mail struct as an argument. It returns
+    /// the string response from the API as JSON.
+    /// 
+    /// ### Example
+    /// ```rust
```

Comment:
Since Rust will run documentation tests, this will fail. We may want to add an `ignore` keyword here.